flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -141,6 +144,25 @@ public boolean isClosureCleanerEnabled() {
 	}
 
 	/**
+	 * Sets the interval of the automatic watermark emission. Watermaks are used throughout
+	 * the streaming system to keep track of the progress of time. They are used, for example,
+	 * for time based windowing.
+	 */
+	public ExecutionConfig setAutoWatermarkInterval(Duration interval) {
```

Comment:
I think it doesn't make sense right now to add documentation for this without a document that describes the broad concepts in Streaming. This touches on a lot of stuff such as event-time vs processing-time, windowing semantics and so on. I think @StephanEwen started the discussion on such a document somewhere.